src/NServiceBus.Metrics.PerformanceCounters/PerformanceCounterUpdater.cs

```diff
@@ -98,10 +90,16 @@ async Task CleanupAndSwallowExceptions(CancellationToken cancellationToken)
                     }
                 }
             }
-        }
-        catch (Exception ex)
-        {
-            logger.Error("Failed to reset performance counter buffers", ex);
+            catch (Exception ex) when (ex.IsCausedBy(cancellationToken))
+            {
+                // private token, updater is being stopped, don't log the exception because the Task.Delay stack trace is not interesting
+                break;
+            }
+            catch (Exception ex)
+            {
+                logger.Error("Failed to reset performance counter buffers", ex);
+                break;
```

Comment:
This preserves the existing behaviour, but it's more obvious now that the loop is broken when a non-cancellation exception is thrown.

Is that desirable? Given the operations that are happening in the `try`, I suspect this (almost) never happens), but it should it not log and continue? The first operation in the `try` is a `Task.Delay`, so it won't be a tight loop.

This outside the scope of what we're doing right now, but I couldn't help noticing it.

Reply:
> Given the operations that are happening in the try, I suspect this (almost) never happens, but it should it not log and continue? The first operation in the try is a Task.Delay, so it won't be a tight loop.

Yea, it makes more sense to me to continue